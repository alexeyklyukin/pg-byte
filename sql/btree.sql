TRUNCATE TABLE test;
CREATE INDEX ON test(data);
SET enable_seqscan to 'off';
INSERT INTO test SELECT id, 'one' FROM generate_series(1, 5000) id;
INSERT INTO test SELECT id, 'two' FROM generate_series(5001, 5002) id;
INSERT INTO test SELECT id, 'three' FROM generate_series(5003, 5004) id;
INSERT INTO test SELECT id, 'four' FROM generate_series(5005, 9999) id;
ANALYZE test;
EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data = 'two';
EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data between 'three' and 'four';
EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data < 'two';
EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data > 'three' and data < 'four';
EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data > 'four';
DROP TABLE test;
RESET enable_seqscan;