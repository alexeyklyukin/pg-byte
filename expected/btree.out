TRUNCATE TABLE test;
CREATE INDEX ON test(data);
SET enable_seqscan to 'off';
INSERT INTO test SELECT id, 'one' FROM generate_series(1, 5000) id;
INSERT INTO test SELECT id, 'two' FROM generate_series(5001, 5002) id;
INSERT INTO test SELECT id, 'three' FROM generate_series(5003, 5004) id;
INSERT INTO test SELECT id, 'four' FROM generate_series(5005, 9999) id;
ANALYZE test;
EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data = 'two';
                           QUERY PLAN                           
----------------------------------------------------------------
 Index Scan using test_data_idx on test (actual rows=2 loops=1)
   Index Cond: (data = 'two'::pg_byte)
(2 rows)

EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data between 'three' and 'four';
                                QUERY PLAN                                
--------------------------------------------------------------------------
 Index Scan using test_data_idx on test (actual rows=4997 loops=1)
   Index Cond: ((data >= 'three'::pg_byte) AND (data <= 'four'::pg_byte))
(2 rows)

EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data < 'two';
                            QUERY PLAN                             
-------------------------------------------------------------------
 Index Scan using test_data_idx on test (actual rows=5000 loops=1)
   Index Cond: (data < 'two'::pg_byte)
(2 rows)

EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data > 'three' and data < 'four';
                               QUERY PLAN                               
------------------------------------------------------------------------
 Index Scan using test_data_idx on test (actual rows=0 loops=1)
   Index Cond: ((data > 'three'::pg_byte) AND (data < 'four'::pg_byte))
(2 rows)

EXPLAIN (ANALYZE, COSTS FALSE, TIMING FALSE, SUMMARY FALSE) SELECT id FROM test WHERE data > 'four';
                           QUERY PLAN                           
----------------------------------------------------------------
 Index Scan using test_data_idx on test (actual rows=0 loops=1)
   Index Cond: (data > 'four'::pg_byte)
(2 rows)

DROP TABLE test;
RESET enable_seqscan;
